### PR TITLE
[GraphQL] Add sort by entity name option

### PR DESCRIPTION
**Pull request by jamesdphillips**
Adds the ability to sort a list events by entity name.

Contributes toward https://github.com/sensu/sensu-enterprise-go/issues/1218

---

Signed-off-by: James Phillips <jamesdphillips@gmail.com>

### CODE REVIEW

**Review by echlebek — APPROVED** [review]
:borat:

**Comment by jamesdphillips** [comment]
@echlebek I'm not super aware of our current process, but will the `api/core` tag have to be bumped after this and #4099 are merged?

**Comment by echlebek** [comment]
@jamesdphillips yes. To complicate matters further, I have an open PR that is tagged as `api/core/v2.5.0-alpha` :)

It might be best to avoid tagging until all the open PRs affecting core/v2 are merged.

File: api/core/v2/event_test.go [hunk 1]
```diff
@@ -539,6 +539,40 @@ func TestEventsByTimestamp(t *testing.T) {
...
+func TestEventsByEntity(t *testing.T) {
+	a1 := FixtureEvent("a", "a")
+	a2 := FixtureEvent("a", "b")
+	b1 := FixtureEvent("b", "a")
+	b2 := FixtureEvent("b", "b")
+
+	testCases := []struct {
+		name     string
+		inEvents []*Event
+		inDir    bool
```
Comment (ccressent) [inline api/core/v2/event_test.go:551]: I don't understand why that's called `inDir`; why not `ascending`?

File: api/core/v2/event.go [hunk 2]
```diff
@@ -215,31 +215,39 @@ func EventsByLastOk(es []*Event) sort.Interface {
...
-	}
-	if b.Entity != nil {
-		bi = b.Entity.Name
-	}
-	if b.Check != nil {
-		bi += b.Check.Name
-	}
+// EventsByEntity can be used to sort a given collection of event by the name
+// of it's referenced entity.
+func EventsByEntity(es []*Event, asc bool) sort.Interface {
```
Comment (ccressent) [inline api/core/v2/event.go:224]: Why not `EventsByEntityName`?

File: api/core/v2/event.go [hunk 3]
```diff
@@ -215,31 +215,39 @@ func EventsByLastOk(es []*Event) sort.Interface {
...
-		ai += a.Check.Name
-	}
-	if b.Entity != nil {
-		bi = b.Entity.Name
-	}
-	if b.Check != nil {
-		bi += b.Check.Name
-	}
+// EventsByEntity can be used to sort a given collection of event by the name
+// of it's referenced entity.
```
Comment (ccressent) [inline api/core/v2/event.go:223]: It seems to me that it sorts on entity name + check name?

**Pull request merged**